### PR TITLE
Delta Lake connector documentation.

### PR DESCRIPTION
**Pull request by ryzhyk**
Is this a user-visible change (yes/no): yes

### CODE REVIEW

**Review by lalithsuresh — APPROVED** [review]

File: docs/connectors/sources/delta.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  * `follow` - follow the changelog of the table, only ingesting changes (new and deleted rows)
+  * `snapshot_and_follow` - Read a snapshot of the table before switching to the `follow` mode.
+    This mode implements the backfill pattern where we load historical data for the table
+    before ingesting the stream of real-time updates.
+
+### Optional parameters
+
+* `timestamp_column` - Table column that serves as an event timestamp.  When this option is
+   specified, and `mode` is one of `snapshot` or `snapshot_and_follow`,
+   the snapshot of the table is ingested in the timestamp order.  This setting is required
```
[the remark this answers is not in the document]
Comment (ryzhyk) [reply docs/connectors/sources/delta.md:29]: Nope, we don't have the plumbing for this.

File: docs/connectors/sources/delta.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+Create a Delta Lake input connector that reads a snapshot of a table from a public S3 bucket.
+Note the `aws_skip_signature` flag, required to read from the bucket without authentcation.  The
+snapshot will be sorted by the `unix_time` column.
+
+```bash
+curl -i -X PUT http://localhost:8080/v0/connectors/delta-input-test \
+-H "Authorization: Bearer <API-KEY>" \
+-H 'Content-Type: application/json' \
+-d '{
```
[the remark this answers is not in the document]
Comment (ryzhyk) [reply docs/connectors/sources/delta.md:89]: Yes, it  reads all columns. We can in theory extend this to allow arbitrary SELECT clauses.

**Comment by ryzhyk** [comment]
@lalithsuresh , I am writing output connector docs now. Will request your review for this PR again when I'm done.

File: docs/connectors/sinks/delta.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+   When not specified, the buffer can grow indefinitely until one of
+   the other trigger conditions is satisfied.
+
+   NOTE: this configuration option requires the `enable_output_buffer` flag
+   to be set.
+
+:::note
+
+when the `enable_output_buffer` flag is set, at least one of
+`max_output_buffer_time_millis` or `max_output_buffer_size_records` must be
```
Comment (lalithsuresh) [inline docs/connectors/sinks/delta.md:81]: Would this misconfiguration return a useful error to the user?

File: docs/connectors/sinks/delta.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    "transport": {
+      "name": "delta_table_output",
+      "config": {
+        "uri": "s3://feldera-fraud-detection-demo/feature_train",
+        "mode": "truncate",
+        "aws_access_key_id": <AWS_ACCESS_KEY_ID>,
+        "aws_secret_access_key": <AWS_SECRET_ACCESS_KEY>,
+        "aws_region": "us-east-1"
+      },
+      "enable_output_buffer": true,
```
Comment (lalithsuresh) [inline docs/connectors/sinks/delta.md:115]: Is the buffer setting common to all output connectors? If so, we need a separate documentation section for it.

**Review by lalithsuresh — APPROVED** [review]

File: docs/connectors/sinks/delta.md (hunk 4 above)
Comment (ryzhyk) [reply docs/connectors/sinks/delta.md:115]: Yes, and yes, but it doesn't make a whole lot of sense for the other connectors we have now. I'll add the section.

File: docs/connectors/sinks/delta.md (hunk 3 above)
Comment (ryzhyk) [reply docs/connectors/sinks/delta.md]: It shoud now.

**Pull request merged**